packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -2644,17 +2650,87 @@ export default abstract class SqlIntegration
         END AS bandit_period`;
   }
 
+  getCovariateMetricCTE({
+    dimensionCols,
+    baseIdType,
+    regressionAdjustedMetrics,
+    baseMetricTable,
+    getCrossFactTableDenominator,
+  }: {
+    dimensionCols: DimensionColumnData[];
+    baseIdType: string;
+    regressionAdjustedMetrics: FactMetricData[];
+    baseMetricTable: string;
+    getCrossFactTableDenominator: boolean;
+  }): string {
+    return `
+          SELECT 
```

Comment:
nit: Should we reduce the indentation here? Or is this because of where this function is called in the bigger SQL statement?

Reply:
Reduced it. It should all get reformatted anyways (except for ClickHouse, where we need a formatter that works).